lib/src/results.dart

```diff
@@ -53,11 +52,9 @@ class RealmResults<T extends Object?> extends collection.IterableBase<T> with Re
     }
   }
 
-  /// `true` if the `Results` collection is empty.
   @override
   bool get isEmpty => length == 0;
 
-  /// Returns a new `Iterator` that allows iterating the elements in this `RealmResults`.
```

Comment:
While `RealmResults` is extending `Iterable` it should not try hiding fact that it is a `RealmResults` object implementation. In this line of thought why not leave the comments mentioning it.